pkg/logging/config_test.go

```diff
@@ -93,6 +95,56 @@ func TestOurConfig(t *testing.T) {
 	}
 }
 
+func TestLogLevelTestConfig(t *testing.T) {
+	wantCfg := `{
+  "level": "debug",
+  "development": false,
+  "outputPaths": ["stdout"],
+  "errorOutputPaths": ["stderr"],
+  "encoding": "json",
+  "encoderConfig": {
+    "timeKey": "ts",
+    "levelKey": "level",
+    "nameKey": "logger",
+    "callerKey": "caller",
+    "messageKey": "msg",
+    "stacktraceKey": "stacktrace",
+    "lineEnding": "",
+    "levelEncoder": "",
+    "timeEncoder": "iso8601",
+    "durationEncoder": "",
+    "callerEncoder": ""
+  }
+}
+`
+	wantLevel := zapcore.DebugLevel
+	components := []string{
+		"autoscaler",
+		"controller",
+		"queueproxy",
+		"webhook",
+		"activator",
+	}
+	cm, _ := ConfigMapsFromTestFile(t, testConfigFileName, "loglevel.autoscaler", "loglevel.controller", "loglevel.queueproxy", "loglevel.webhook", "loglevel.activator", "zap-logger-config")
+	cfg, err := logging.NewConfigFromConfigMap(cm)
+
+	if err != nil {
+		t.Errorf("Expected no errors. got: %v", err)
+	}
+	if cfg == nil {
+		t.Errorf("NewConfigFromConfigMap(actual) = %v, want non-nil", cfg)
+	}
+
+	for _, c := range components {
+		if got := cfg.LoggingLevel[c]; got != wantLevel {
+			t.Errorf("LoggingLevel[%q] = %v, want %v", c, got, wantLevel)
+		}
+	}
+	if got := cfg.LoggingConfig; got != wantCfg {
+		t.Errorf("LoggingConfig = %v, want %v, diff %s", got, wantCfg, cmp.Diff(got, wantCfg))
```

Comment:
```suggestion
		t.Errorf("LoggingConfig = %v, want %v, diff(-want +got) %s", got, wantCfg, cmp.Diff(wantCfg, got))
```
(https://godoc.org/github.com/google/go-cmp/cmp#Diff, most (but surely not all) follows this, see the `Diff` example)